clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1511,6 +1561,9 @@ void Sema::MarkDevice(void) {
               Diag(Attr->getLocation(), diag::note_conflicting_attribute);
               SYCLKernel->setInvalidDecl();
             }
+          } else if (KBSimdAttr && (Attr->getSubGroupSize() != 1)) {
+            // ESIMD kernel can't call functions with reqd_subg_group_size != 1.
+            reportConflictingAttrs(*this, KernelBody, KBSimdAttr, Attr);
```

Comment:
Typically we diagnose conflicts like this in SemaDeclAttr when we first encounter these, is there a good reason to not do it the typical way?

Reply:
I don't know how it is typically diagnosed. I'm following the existing approach right above and below. How would I know which is the right one - SemaDeclAttr or this one?

Reply:
Not sure that we can do this diagnostic in SemaDeclAttr, this can be too early since sycl kernels appear as parameters of template functions.

Reply:
SemaDeclAttr is typically where we diagnose ALL attributes as we get to them, but @Fznamznon is probably right here, i think this is for things that happen AFTER they have been redistributed.